prometheus/go_collector.go

```diff
@@ -236,6 +242,9 @@ func (c *goCollector) Describe(ch chan<- *Desc) {
 func (c *goCollector) Collect(ch chan<- Metric) {
 	c.goroutines.Set(float64(runtime.NumGoroutine()))
 	ch <- c.goroutines
+	n, _ := runtime.ThreadCreateProfile(nil)
+	c.threads.Set(float64(n))
```

Comment:
I am not sure too, but I think it is helpful in some way and I need some time or experiment to dig it out, and add a threads number may be the first step.